frontend/src/presentation/pages/CollaborationRoomPage.tsx

```diff
@@ -2,16 +2,20 @@ import React, { useState, useCallback, useEffect, useRef } from "react";
 import styles from "./CollaborationRoomPage.module.css";
 import CodeEditor from "presentation/components/CodeEditor/CodeEditor";
 import { QuestionDetail } from "presentation/components/QuestionDetail";
-import { initialQuestions } from "data/repositories/mockQuestionRepository";
-import { useParams } from "react-router-dom";
+import { useLocation } from "react-router-dom";
 import { useResizable } from "react-resizable-layout";
 import NotFound from "./NotFound";
 import { OutputBox } from "presentation/components/CodeEditor/OutputBox";
 import ToggleButton from "presentation/components/buttons/ToggleButton";
 import ChatFrame from "presentation/components/iframe/ChatFrame";
+import { Question } from "domain/entities/Question";
+import { questionUseCases } from "domain/usecases/QuestionUseCases";
 
 const CollaborationRoomPage: React.FC = () => {
-    const { roomId } = useParams();
+    const location = useLocation();
+    const { message, category, difficulty, roomId, attemptStartedAt, matchId, matchUserId, questionId } = location.state;
```

Comment:
Does this handle the case where the user reloads the page? I'm not sure if the state persists across page reloads

Reply:
It does. At least, it does when I reload. Hell, the state persists when I leave the room via the leave room button, then I go back via the browser's back button.